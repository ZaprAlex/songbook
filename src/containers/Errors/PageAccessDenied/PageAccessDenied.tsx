import React, { useEffect } from 'react';
import { useLocation } from 'react-router-dom';
import isEmpty from 'lodash/isEmpty';

import { noop } from '../../../utils/helper';
import { stateLocationType } from '../../../utils/navigation/types';
import { IButton } from '../../../component/Button';
import InformContainer from '../../../component/InformContainer';
import icon from '../../../static/images/empty.png';

import styles from '../Errors.module.css';

export interface PageAccessDeniedProps {
    header?: string;
    message?: string;
    buttonLabel?: string;
    onClick?: () => void;
    onMount?: () => void;
}

const PageAccessDenied: React.FC<PageAccessDeniedProps> = ({
    onClick,
    header = 'Access denied',
    message,
    buttonLabel,
    onMount = noop,
}) => {
    const { state }: stateLocationType = useLocation();
    const buttonProps: IButton = { size: 'x-large' };

    if (!isEmpty(state)) {
        onClick = state.onClick ? state.onClick : onClick;
        header = state.header ? state.header : header;
        message = state.message ? state.message : message;
        buttonLabel = state.buttonLabel ? state.buttonLabel : buttonLabel;
    }

    useEffect(onMount, []);

    return (
        <div className={styles.page}>
            <InformContainer
                icon={icon}
                header={header}
                message={message}
                contentClassName={styles.content}
                {...(buttonLabel && onClick
                    ? {
                          buttonLabel,
                          onClick,
                          buttonClassName: styles.button,
                          buttonProps,
                      }
                    : {})}
            />
        </div>
    );
};

export default PageAccessDenied;
