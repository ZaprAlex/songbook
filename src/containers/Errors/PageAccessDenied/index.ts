export { default } from './PageAccessDenied';
export * from './PageAccessDenied';
