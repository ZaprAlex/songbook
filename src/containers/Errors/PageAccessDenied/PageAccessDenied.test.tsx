import React from 'react';
import { Router } from 'react-router-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { MemoryHistory } from 'history/createMemoryHistory';

import PageAccessDenied, { PageAccessDeniedProps } from './PageAccessDenied';

describe('PageAccessDenied', () => {
    let history: MemoryHistory;
    const header = 'TestHeader';
    const message = 'TestMessage';
    const buttonLabel = 'TestButtonLabel';
    const onClick = jest.fn();
    const initialProps: PageAccessDeniedProps = {
        header,
    };

    const component = (additionalProps?: PageAccessDeniedProps) =>
        render(
            <Router history={history}>
                <PageAccessDenied {...initialProps} {...additionalProps} />
            </Router>
        );

    beforeEach(() => {
        jest.clearAllMocks();
        jest.restoreAllMocks();
        history = createMemoryHistory();
    });

    test('w/o message and btn', () => {
        const { container } = component();

        expect(screen.queryByText(header)).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });

    test('with message and button', () => {
        const { container } = component({ message, buttonLabel, onClick });

        expect(container.firstChild).toMatchSnapshot();

        expect(screen.queryByText(message)).toBeInTheDocument();
        expect(screen.queryByText(buttonLabel)).toBeInTheDocument();

        fireEvent.click(screen.getByText(buttonLabel));

        expect(onClick).toBeCalled();
    });
});
