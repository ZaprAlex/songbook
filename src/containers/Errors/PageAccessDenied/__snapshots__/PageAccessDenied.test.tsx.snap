// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageAccessDenied w/o message and btn 1`] = `
<div
  class="page"
>
  <div
    class="content content"
  >
    <div
      class="iconWrapper"
    >
      <img
        alt=""
        class="icon"
        data-test-id="inform-container-icon"
        src="cat-empty.png"
      />
    </div>
    <div
      class="header"
    >
      TestHeader
    </div>
    <div
      class="buttonBlock"
    />
  </div>
</div>
`;

exports[`PageAccessDenied with message and button 1`] = `
<div
  class="page"
>
  <div
    class="content content"
  >
    <div
      class="iconWrapper"
    >
      <img
        alt=""
        class="icon"
        data-test-id="inform-container-icon"
        src="cat-empty.png"
      />
    </div>
    <div
      class="header"
    >
      TestHeader
    </div>
    <div
      class="message"
    >
      TestMessage
    </div>
    <div
      class="buttonBlock"
    >
      <button
        class="btn btn-green btn-contained btn-x-large btnSkeuomorph button"
        data-test-id="button-submit"
        type="button"
      >
        <span
          class="buttonLabel"
        >
          TestButtonLabel
        </span>
      </button>
    </div>
  </div>
</div>
`;
